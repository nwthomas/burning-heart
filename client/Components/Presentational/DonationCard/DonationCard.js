import React from "react";
import { Text, View, StyleSheet } from "react-native";

const DonationCard = ({ donation }) => {
  return (
    <View style={styles.cardContainer}>
      <Text style={styles.donationCardText}>{donation.charityName}</Text>
      <Text style={styles.donationAmount}>{`$${parseFloat(
        donation.amount / 100
      ).toFixed(2)}`}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  cardContainer: {
    alignSelf: "stretch",
    paddingTop: 30,
    paddingBottom: 30,
    borderBottomColor: "#cacacf",
    borderBottomWidth: 3
  },
  donationCardText: {
    fontFamily: "Montserrat-Medium",
    fontSize: 16
  },
  donationAmount: {
    fontFamily: "OpenSans-Italic",
    fontSize: 14,
    marginTop: 5
  }
});

export default DonationCard;
