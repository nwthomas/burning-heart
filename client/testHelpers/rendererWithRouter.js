import React from "react";
import { NativeRouter as Router } from "react-router-native";
import renderer from "react-test-renderer";
import { createMemoryHistory } from "history";

const rendererWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...renderer.create(<Router history={history}>{ui}</Router>),
    history
  };
};

export default rendererWithRouter;
