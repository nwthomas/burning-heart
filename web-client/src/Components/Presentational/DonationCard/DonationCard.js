import React from "react";

const DonationCard = props => {
  return (
    <div className="donation__card">
      <p className="donation__title">{props.donation.charityName}</p>
      <p className="donation__amount">{`$${props.donation.amount / 100}`}</p>
    </div>
  );
};

export default DonationCard;
