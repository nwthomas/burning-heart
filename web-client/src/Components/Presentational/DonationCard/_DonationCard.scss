.donation__card {
  width: 90%;
  border-bottom: 2px solid #cacacf;
  height: 100px;
  margin: 0 5%;
  display: flex;
  justify-content: center;
  flex-direction: column;

  .donation__title {
    margin-bottom: 10px;
  }
}

.donation__card:last-of-type {
  border-bottom: 0;
}
