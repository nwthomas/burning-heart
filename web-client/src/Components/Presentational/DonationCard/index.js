export { default as DonationCard } from "./DonationCard";
