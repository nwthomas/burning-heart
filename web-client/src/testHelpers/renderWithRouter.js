import React from "react";
import { Router } from "react-router-dom";
import { render } from "react-testing-library";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { store } from "../store";

const renderWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...render(
      <Router history={history}>
        <Provider store={store}>{ui}</Provider>
      </Router>
    ),
    history
  };
};

export default renderWithRouter;
