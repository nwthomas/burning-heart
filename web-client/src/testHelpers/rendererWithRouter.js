import React from "react";
import { Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { store } from "../store";

const rendererWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...renderer.create(
      <Router history={history}>
        <Provider store={store}>{ui}</Provider>
      </Router>
    ),
    history
  };
};

export default rendererWithRouter;
