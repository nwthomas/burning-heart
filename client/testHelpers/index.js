import renderWithRouter from "./renderWithRouter";
import rendererWithRouter from "./rendererWithRouter";

export { renderWithRouter, rendererWithRouter };
