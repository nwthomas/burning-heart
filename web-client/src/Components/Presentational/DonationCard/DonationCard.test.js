import React from "react";
import "react-testing-library/cleanup-after-each";
import "jest-dom/extend-expect";
import { renderWithRouter, rendererWithRouter } from "../../../testHelpers";

import DonationCard from "./DonationCard";

describe("<DonationCard />", () => {
  it("should render the DonationCard component without crashing", () => {
    const donation = { charityName: "Name", amount: 1000 };
    const helpers = renderWithRouter(<DonationCard donation={donation} />);
  });

  it("should match the snapshot of DonationCard", () => {
    const donation = { charityName: "Name", amount: 1000 };
    const tree = rendererWithRouter(<DonationCard donation={donation} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
