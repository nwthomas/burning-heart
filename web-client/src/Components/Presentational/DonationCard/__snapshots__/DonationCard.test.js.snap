// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DonationCard /> should match the snapshot of DonationCard 1`] = `
<div
  className="donation__card"
>
  <p
    className="donation__title"
  >
    Name
  </p>
  <p
    className="donation__amount"
  >
    $10
  </p>
</div>
`;
